src/mongocrypt-buffer-private.h

```diff
@@ -209,4 +209,12 @@ _mongocrypt_buffer_steal_from_string (_mongocrypt_buffer_t *buf,
 void
 _mongocrypt_buffer_copy_from_uint64_le (_mongocrypt_buffer_t *buf, uint64_t value);
 
+/* _mongocrypt_buffer_view initializes @out as a writable view of the data from
+ * @in. Returns false on error. */
+bool
+_mongocrypt_buffer_view (_mongocrypt_buffer_t *out,
+                         const _mongocrypt_buffer_t *in,
+                         uint32_t offset,
+                         uint32_t len) MONGOCRYPT_WARN_UNUSED_RESULT;
```

Comment:
```suggestion
/* _mongocrypt_buffer_from_view initializes @out as a non-owning buffer to a
 * range of data from @in specified by @offset and @len. Returns false on error.
 */
bool
_mongocrypt_buffer_from_subrange (_mongocrypt_buffer_t *out,
                                  const _mongocrypt_buffer_t *in,
                                  uint32_t offset,
                                  uint32_t len) MONGOCRYPT_WARN_UNUSED_RESULT;
```
I like this addition. However, `_mongocrypt_buffer_t` does not support const correctness, only ownership correctness (there is no read-only "view" mode for `_mongocrypt_buffer_t` objects). I suggest these slight changes to name and description to avoid confusion with the interface of a possible const-correct alternative to `_mongocrypt_buffer_t`.

Reply:
Good idea. Will do.